shopfloor/actions/search.py

```diff
@@ -31,11 +32,27 @@ def package_from_scan(self, barcode):
             return model.browse()
         return model.search([("name", "=", barcode)], limit=1)
 
-    def picking_from_scan(self, barcode):
+    def picking_from_scan(self, barcode, use_origin=False):
```

Comment:
This should be restricted to the picking types defined on the menu

Reply:
This wasn't the case in the previous version either right?

This check for picking types is done in each scenario's domain methods, as far as I can tell.

Reply:
Yes, this method is generic. You can validate the result where/when you need.

Reply:
As it was before, it was fine because it was searching on the name and it's unique.
When searching on the origin, it may cause issue. The method was designed to return a single picking which is not the case anymore.
You can add a fixme for later but I would limit the last search to 1.

Reply:
This has been dealt with in the reception scenario as well (displaying the filtered list of pickings if there's more than 1 returned). That's why we allow more than 1 picking if we search by origin.

I think it should be each scenario that takes care of dealing with this case when filtering by origin.